spring-integration-file/src/test/java/org/springframework/integration/file/tail/FileTailingMessageProducerTests.java

```diff
@@ -124,9 +127,47 @@ public void canRecalculateCommandWhenFileOrOptionsChanged() throws IOException {
 		adapter.stop();
 	}
 
+	@Test
+	@TailAvailable
+	public void testIdleEvent() throws Exception {
+		OSDelegatingFileTailingMessageProducer adapter = new OSDelegatingFileTailingMessageProducer();
+		adapter.setOptions(TAIL_OPTIONS_FOLLOW_NAME_ALL_LINES);
+		ThreadPoolTaskScheduler taskScheduler = new ThreadPoolTaskScheduler();
+		taskScheduler.afterPropertiesSet();
+		adapter.setTaskScheduler(taskScheduler);
+		final List<FileTailingIdleEvent> events = new ArrayList<>();
+		adapter.setApplicationEventPublisher(event -> {
+			if (event instanceof FileTailingIdleEvent) {
+				FileTailingIdleEvent tailEvent = (FileTailingIdleEvent) event;
+				logger.warn(event);
```

Comment:
I'm not sure in the WARN.
 Let remove it or at least convert it into DEBUG. 
I see the place you borrowed an approach and that doesn't look correct for me, too.
Let's change that to DEBUG as well!
My concern to avoid extra logging message, when we can live without them.
If there is some problem with test, we always can switch on DEBUG level for the category in the logging config.